packages/shared/lib/clients/orchestrator.ts

```diff
@@ -857,4 +880,247 @@ export class Orchestrator {
+                        connectionDetails: nangoConnection,
+                        providerConfig,
+                        syncName,
+                        sync,
+                        syncData
+                    }
+                });
+
+                await updateSuccessActivityLog(activityLogId, false);
+
+                return Err(interval.error);
+            }
+
+            const schedule = await this.client.recurring({
+                name: this.getScheduleName({ environmentId: nangoConnection.environment_id, syncId: sync.id }),
+                state: syncData.auto_start ? 'STARTED' : 'PAUSED',
+                frequencyMs: ms(interval.value as StringValue),
+                groupKey: 'sync',
+                retry: { count: 0, max: 3 },
+                timeoutSettingsInSecs: {
+                    createdToStarted: 60 * 60, // 1 hour
+                    startedToCompleted: 60 * 60 * 24, // 1 day
+                    heartbeat: 30 * 60 // 30 minutes
+                },
+                startsAt: new Date(),
+                args: {
+                    type: 'sync',
+                    syncId: sync.id,
+                    syncName,
+                    debug,
+                    connection: {
+                        id: nangoConnection.id!,
+                        provider_config_key: nangoConnection.provider_config_key,
+                        environment_id: nangoConnection.environment_id,
+                        connection_id: nangoConnection.connection_id
+                    }
+                }
+            });
+
+            if (schedule.isErr()) {
+                throw schedule.error;
+            }
+
+            await createActivityLogMessageAndEnd({
+                level: 'info',
+                environment_id: nangoConnection.environment_id,
+                activity_log_id: activityLogId,
+                content: `Scheduled to run "${syncData.runs}"`,
+                timestamp: Date.now()
+            });
+            await logCtx.info('Scheduled successfully', { runs: syncData.runs });
+            await logCtx.success();
+            return Ok(undefined);
+        } catch (err) {
+            errorManager.report(err, {
+                source: ErrorSourceEnum.PLATFORM,
+                operation: LogActionEnum.SYNC_CLIENT,
+                environmentId: nangoConnection.environment_id,
+                metadata: {
+                    syncName,
+                    connectionDetails: JSON.stringify(nangoConnection),
+                    syncId: sync.id,
+                    providerConfig,
+                    syncData: JSON.stringify(syncData)
+                }
+            });
+            if (logCtx) {
+                await logCtx.error('Failed to init sync', { error: err });
+                await logCtx.failed();
+            }
+            return Err(`Failed to schedule sync: ${err}`);
+        }
+    }
+    // TODO: remove once temporal is removed
+    async scheduleSyncHelper(
```

Comment:
same as with `runSyncCommand`. This function responsibility is to manage the flag related logic